### PR TITLE
Fixes memory allocation in nested JSON tokenizer 

### PR DESCRIPTION
**Pull request by elstehle**
## Description

This PR makes sure that the buffer allocated for the tokens being output by the JSON tokenizer is always sufficiently large for a given input. 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/src/io/json/nested_json_gpu.cu [hunk 1]
```diff
@@ -1117,9 +1117,18 @@ std::pair<rmm::device_uvector<PdaTokenT>, rmm::device_uvector<SymbolOffsetT>> ge
                                        stream};
 
   // Perform a PDA-transducer pass
+  // Compute the maximum amount of tokens that can possibly be emitted for a given input size
+  // Worst case ratio of tokens per input char is given for a struct with an empty field name, that
+  // may be arbitrarily deeply nested: {"":_}, where '_' is a placeholder for any JSON value,
+  // possibly another such struct. That is, 6 tokens for 5 chars (plus chars and tokens of '_')
+  std::size_t const min_chars_per_struct  = 5;
+  std::size_t const max_tokens_per_struct = 6;
```
[the remark this answers is not in the document]
Comment (elstehle) [reply cpp/src/io/json/nested_json_gpu.cu:1125]: That would be great, but I'm not aware of an algorithm that, given an FST, reliably computes the maximum output size (in acceptable time). Any suggestions?

[the remark this answers is not in the document]
Comment (karthikeyann) [reply cpp/src/io/json/nested_json_gpu.cu]: That's right. It depends on the transitions. I am not aware of such algorithm.

**Review by karthikeyann — APPROVED** [review]

**Review by vuule — COMMENTED** [review]
What a test! :D
Looks good, just one suggestion.

**Review by vuule — APPROVED** [review]

**Comment by elstehle** [comment]
@gpucibot merge

**Pull request merged**